src/shared/dissect-image.c

```diff
@@ -751,6 +785,20 @@ static int dissect_image(
         if (!b)
                 return -ENOMEM;
 
+        r = allowed_fstypes(&fstypes);
+        if (r < 0)
+                return r;
+
+        errno = 0;
+        r = blkid_probe_filter_superblocks_type(b, BLKID_FLTR_ONLYIN, fstypes);
+        if (r != 0)
+                return errno_or_else(EINVAL);
+
+        errno = 0;
+        r = blkid_probe_filter_superblocks_usage(b, BLKID_FLTR_NOTIN, BLKID_USAGE_RAID);
+        if (r != 0)
+                return errno_or_else(EINVAL);
```

Comment:
The same pattern also exist in probe_filesystem_full(). Maybe better to introduce a tiny helper function about that.